server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/ChannelTransport.java

```diff
@@ -39,22 +38,22 @@
 
     final ChannelFactory factory;
 
-    public ChannelTransport(Channel channel, ChannelFactory factory) {
+    public ChannelTransport(JChannel channel, ChannelFactory factory) {
         super(channel);
         this.factory = factory;
     }
 
     @Override
     protected void initRPCDispatcher() {
-        this.dispatcher = new CommandAwareRpcDispatcher(channel, this, globalHandler, this.getTimeoutExecutor(), timeService);
-        MarshallerAdapter adapter = new MarshallerAdapter(this.marshaller) {
-            @Override
-            public Object objectFromBuffer(byte[] buffer, int offset, int length) throws Exception {
-                return ChannelTransport.this.factory.isUnknownForkResponse(ByteBuffer.wrap(buffer, offset, length)) ? CacheNotFoundResponse.INSTANCE : super.objectFromBuffer(buffer, offset, length);
-            }
-        };
-        this.dispatcher.setRequestMarshaller(adapter);
-        this.dispatcher.setResponseMarshaller(adapter);
+        this.dispatcher = new CommandAwareRpcDispatcher(channel, this, globalHandler, this.getTimeoutExecutor(), timeService,
+              marshaller);
+        Field bufField;
```

Comment:
It was supposed to be used by the marshaller adapter, but it's no longer necessary.